host/namespace_delete_test.go

```diff
@@ -0,0 +1,245 @@
+		operatorClient operatorservice.OperatorServiceClient
+		esClient       esclient.IntegrationTestsClient
+
+		cluster       *TestCluster
+		clusterConfig *TestClusterConfig
+		logger        log.Logger
+	}
+)
+
+func TestNamespaceSuite(t *testing.T) {
+	flag.Parse()
+	suite.Run(t, &namespaceTestSuite{})
+}
+
+func dynamicConfig() map[dynamicconfig.Key]interface{} {
+	return map[dynamicconfig.Key]interface{}{
+		dynamicconfig.DeleteNamespaceDeleteActivityRPS: 1000,
+	}
+}
+
+func (s *namespaceTestSuite) SetupSuite() {
+	s.logger = log.NewTestLogger()
+
+	clusterConfig, err := GetTestClusterConfig("testdata/integration_namespace_cluster.yaml")
+	s.Require().NoError(err)
+	s.clusterConfig = clusterConfig
+	clusterConfig.DynamicConfigOverrides = dynamicConfig()
+
+	// Elasticsearch is needed to test advanced visibility code path in reclaim resources workflow.
+	s.esClient = CreateESClient(&s.Suite, clusterConfig.ESConfig, s.logger)
+	PutIndexTemplate(&s.Suite, s.esClient, fmt.Sprintf("testdata/es_%s_index_template.json", clusterConfig.ESConfig.Version), "test-visibility-template")
+	CreateIndex(&s.Suite, s.esClient, clusterConfig.ESConfig.GetVisibilityIndex())
+
+	cluster, err := NewCluster(clusterConfig, s.logger)
+	s.Require().NoError(err)
+	s.cluster = cluster
+	s.frontendClient = s.cluster.GetFrontendClient()
+	s.adminClient = s.cluster.GetAdminClient()
+	s.operatorClient = s.cluster.GetOperatorClient()
+}
+
+func (s *namespaceTestSuite) TearDownSuite() {
+	s.cluster.TearDownCluster()
+	DeleteIndex(&s.Suite, s.esClient, s.clusterConfig.ESConfig.GetVisibilityIndex())
+}
+
+func (s *namespaceTestSuite) SetupTest() {
+	// Have to define our overridden assertions in the test setup. If we did it earlier, s.T() will return nil
+	s.Assertions = require.New(s.T())
+}
+
+func (s *namespaceTestSuite) Test_NamespaceDelete_Empty() {
+	ctx, cancel := rpc.NewContextWithTimeoutAndVersionHeaders(10000 * time.Second)
+	defer cancel()
+
+	retention := 24 * time.Hour
+	_, err := s.frontendClient.RegisterNamespace(ctx, &workflowservice.RegisterNamespaceRequest{
+		Namespace:                        "ns_name_san_diego",
+		Description:                      "Namespace to delete",
+		WorkflowExecutionRetentionPeriod: &retention,
+		HistoryArchivalState:             enumspb.ARCHIVAL_STATE_DISABLED,
+		VisibilityArchivalState:          enumspb.ARCHIVAL_STATE_DISABLED,
+	})
+	s.NoError(err)
+	// DescribeNamespace reads directly from database but namespace validator uses cache.
+	s.cluster.RefreshNamespaceCache()
+
+	descResp, err := s.frontendClient.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: "ns_name_san_diego",
+	})
+	s.NoError(err)
+	nsID := descResp.GetNamespaceInfo().GetId()
+
+	delResp, err := s.operatorClient.DeleteNamespace(ctx, &operatorservice.DeleteNamespaceRequest{
+		Namespace: "ns_name_san_diego",
+	})
+	s.NoError(err)
+	s.Equal("ns_name_san_diego-deleted-"+nsID[:5], delResp.GetDeletedNamespace())
```

Comment:
:slightly_frowning_face: to duplicating the renaming logic, but not a big deal

Reply:
There is no workaround for this. There is no way to create namespace with specific Id.